src/cfnlint/data/ExtendedSpecs/all/03_value_types.json

```
@@ -698,6 +698,24 @@
           "PublicReadWrite"
         ]
       },
+      "S3BucketInventoryId": {
+        "AllowedPattern": "[a-zA-Z0-9.-_]+",
+        "AllowedPatternRegex": "^[a-zA-Z0-9.\\-_]+$",
```

["Name can only contain letters, numbers, underscores, periods, or dashes"](https://github.com/aws-cloudformation/cfn-python-lint/issues/1550)

Gonna move the literal `-` somewhere other than in between those other literal characters so it's a bit clearer and doesn't need to be double-escaped

```suggestion
        "AllowedPatternRegex": "^[a-zA-Z0-9-_.]+$",
```